sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/Storage/BlobsCheckpointStore.cs

```diff
@@ -247,37 +308,130 @@ await foreach (BlobItem blob in ContainerClient.GetBlobsAsync(traits: BlobTraits
         /// </summary>
         ///
         /// <param name="checkpoint">The checkpoint containing the information to be stored.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
         ///
         /// <returns>A task to be resolved on when the operation has completed.</returns>
         ///
-        public override async Task UpdateCheckpointAsync(Checkpoint checkpoint)
+        public override Task UpdateCheckpointAsync(Checkpoint checkpoint,
+                                                   CancellationToken cancellationToken)
         {
-            var blobName = string.Format(CheckpointPrefix + checkpoint.PartitionId, checkpoint.FullyQualifiedNamespace.ToLower(), checkpoint.EventHubName.ToLower(), checkpoint.ConsumerGroup.ToLower());
+            cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
 
-            BlobClient blobClient = ContainerClient.GetBlobClient(blobName);
+            var blobName = string.Format(CheckpointPrefix + checkpoint.PartitionId, checkpoint.FullyQualifiedNamespace.ToLower(), checkpoint.EventHubName.ToLower(), checkpoint.ConsumerGroup.ToLower());
+            var blobClient = ContainerClient.GetBlobClient(blobName);
 
             var metadata = new Dictionary<string, string>()
-                {
-                    { BlobMetadataKey.Offset, checkpoint.Offset.ToString() },
-                    { BlobMetadataKey.SequenceNumber, checkpoint.SequenceNumber.ToString() }
-                };
+            {
+                { BlobMetadataKey.Offset, checkpoint.Offset.ToString() },
+                { BlobMetadataKey.SequenceNumber, checkpoint.SequenceNumber.ToString() }
+            };
+
+            Func<CancellationToken, Task> updateCheckpointAsync = async updateCheckpointToken =>
+            {
+                using var blobContent = new MemoryStream(new byte[0]);
```

Comment:
nit:  I'd recommend using `Array.Empty<byte>()` here, as it has the potential to eliminate the allocation.